cloudinit/reporting/events.py

```diff
@@ -81,18 +81,23 @@ def as_dict(self):
         return data
 
 
-def report_event(event):
-    """Report an event to all registered event handlers.
+def report_event(event, excluded_handlers=None):
+    """Report an event to all registered event handlers
+    except the excluded_handlers.
 
     This should generally be called via one of the other functions in
     the reporting module.
 
+    :param excluded_handlers:
+         List of handlers to exclude from reporting the event to.
     :param event_type:
         The type of the event; this should be a constant from the
         reporting module.
     """
-    for _, handler in instantiated_handler_registry.registered_items.items():
-        handler.publish_event(event)
+    handlers = instantiated_handler_registry.registered_items.items()
+    for handler_key, handler in handlers:
+        if not excluded_handlers or handler_key not in excluded_handlers:
+            handler.publish_event(event)
```

Comment:
Sorry about this nit review, but the double negative logic hurts my simple brain and makes me think too hard.

What do you think about this?
```suggestion
        if excluded_handlers and handler_key in excluded_handlers:
            continue    # skip this excluded handler
        handler.publish_event(event)
```

Reply:
Works for me.